prost-build/src/code_generator.rs

```diff
@@ -310,10 +310,15 @@ impl<'a> CodeGenerator<'a> {
         let ty = self.resolve_type(&field, fq_message_name);
 
         let boxed = !repeated
-            && (type_ == Type::Message || type_ == Type::Group)
-            && self
-                .message_graph
-                .is_nested(field.type_name(), fq_message_name);
+            && ((type_ == Type::Message || type_ == Type::Group)
+                && self
+                    .message_graph
+                    .is_nested(field.type_name(), fq_message_name))
+            || (self
+                .config
+                .boxed
+                .get_first_field(&fq_message_name, field.name())
+                .is_some());
```

Comment:
Could we move this to a fn since this is getting a bit complicated now and we reuse it below as well.. I think a fn + a doc comment to make it easier for future readers would go a long way here.